src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -971,11 +971,13 @@ public override int GetHashCode()
             AssertValid();
 
             if (_bits == null)
-                return _sign;
-            int hash = _sign;
+                return HashCode.Combine(_sign);
```

Comment:
Is the extra call to `HashCode.Combine` worth it here?  The existing code just returned `_sign`, since that is the value when `_bits` is null.

Reply:
From @morganbr:

```
            // Provide a way of diffusing bits from something with a limited
            // input hash space. For example, many enums only have a few
            // possible hashes, only using the bottom few bits of the code. Some
            // collections are built on the assumption that hashes are spread
            // over a larger space, so diffusing the bits may help the
            // collection work more efficiently.
```

Reply:
I don't think I follow this logic.  `BigInteger`'s value, when `_bits` is null, is just the `int` value.  Are we doing the same change to `Int32`'s GetHashCode()?

---
In reply to: [167421134](https://github.com/dotnet/corefx/pull/25987#discussion_r167421134) [](ancestors = 167421134)